clients/command-kafka/src/main/java/org/eclipse/hono/client/command/kafka/KafkaBasedInternalCommandConsumer.java

```diff
@@ -196,13 +229,31 @@ public final KafkaBasedInternalCommandConsumer setMetricsSupport(final KafkaClie
         adminClient.createTopics(List.of(newTopic))
                 .all()
                 .whenComplete((v, ex) -> {
-                    context.runOnContext(v1 -> Optional.ofNullable(ex).ifPresentOrElse(promise::fail, promise::complete));
+                    context.runOnContext(v1 -> Optional.ofNullable(ex)
+                            .filter(e -> !(e instanceof TopicExistsException))
+                            .ifPresentOrElse(promise::fail, promise::complete));
                 });
         return promise.future()
                 .onSuccess(v -> LOG.debug("created topic [{}]", topicName))
                 .onFailure(thr -> LOG.error("error creating topic [{}]", topicName, thr));
     }
 
+    private Future<Void> createTopicWithRetries() {
```

Comment:
Retries should be stopped when `stop()` has been called.
Also there's no need to try to delete the topic in `stop()` if `isTopicCreated` is `false`.

Reply:
> Retries should be stopped when `stop()` has been called.

How about setting `retryCreateTopic.set(false)` in `stop()` to ensure this?